examples/earth-and-mars/index.js

```diff
@@ -0,0 +1,202 @@
+const options = {
+    numPoints: 1e5,
+    chartType: 'WebGL',
+    showMultiColor: false
+};
+
+let data = [];
+let currentData;
+
+const worker = new Worker('worker.js');
+worker.onmessage = e => {
+    data = e.data;
+    currentData = data.slice(0, options.numPoints);
+
+    x.domain([currentData[0].date, currentData[currentData.length - 1].date]);
+    x2 = x.copy();
+
+    render();
+};
+
+const x = d3.scaleTime();
+const y = d3.scaleLinear();
+
+y.domain([0, 400]);
+let x2 = x.copy();
+
+const crossValue = d => d.date;
+const mainValue = d => d.distance;
+
+const blueGradientDecorate = program => {
+    program.fragmentShader().appendBody(`
+        vec4 color = vec4(70.0/255.0, 130.0/255.0, 180.0/255.0, 1.0);
+        float colorModifier = 1.0 - gl_FragCoord.y / 421.0;
+        colorModifier = smoothstep(0.875, 1.0, colorModifier);
+        gl_FragColor = vec4(color + ((vec4(1.0, 1.0, 1.0, 1.0) - color) * colorModifier));
+    `);
+};
+
+const webglArea = fc
+    .seriesWebglArea()
+    .mainValue(mainValue)
+    .crossValue(crossValue)
+    .defined(() => true)
```

Comment:
Does defined have a default value? I think it probably should.

Reply:
`defined` has a default implementation that involves a `for` loop. This optimises the defined check so we don't run a loop for every data point